### PR TITLE
Convert `test/otel` into a module

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

This PR changes the fixtures in `test/otel` to real files in order to include this test module in the agent go mod checks. 

### Motivation

The otel integration test broke when [7.56.0-rc.1 was being built](https://dd.slack.com/archives/CE4PS8MHS/p1720448103494119?thread_ts=1720441475.151049&cid=CE4PS8MHS) because `go.mod.fake` was not up to date. The reason why we made these `.fake` in the first place is because we wanted to avoid adding upstream OTel imports, but we no longer plan on doing this and only import datadog-agent modules in test/otel.

### CODE REVIEW

File: tasks/modules.py [hunk 1]
```diff
@@ -278,6 +278,7 @@ def dependency_path(self, agent_version):
         targets=["./pkg/runner", "./pkg/utils/e2e/client"],
         lint_targets=[".", "./examples"],  # need to explicitly list "examples", otherwise it is skipped
     ),
+    "test/otel": GoModule("test/otel", independent=True, used_by_otel=True),
```
Comment (pducolin) [inline tasks/modules.py:281]: **❓ question**
Is this used by `otel`?

Comment (liustanley) [reply tasks/modules.py:281]: Technically not, we can remove `used_by_otel` if you prefer. I just added it to include it in the otel module version check.

**Review by dineshg13 — APPROVED** [review]

**Review by CelianR — APPROVED** [review]

File: tasks/modules.py (hunk 1 above)
Comment (pducolin) [reply tasks/modules.py:281]: Thank you, then I would rename `used_by_otel` to `include_in_otel_module_check`, or something similar.

**Review by pducolin — APPROVED** [review]

File: tasks/modules.py (hunk 1 above)
Comment (liustanley) [reply tasks/modules.py:286]: `used_by_otel` is also used by ASC in [update_go_mods](https://github.com/DataDog/datadog-agent/blob/5998a28ed6a4f0f96ef7152ff0093c8b53c0ef64/tasks/update_go.py#L191), so I'm inclined to keep this name. We can change it in a separate PR if preferred though

**Pull request merged**